core/core.go

```diff
@@ -367,6 +367,9 @@ func (n *IpfsNode) SetupOfflineRouting() error {
 	}
 
 	n.Routing = offroute.NewOfflineRouter(n.Repo.Datastore(), n.PrivateKey)
+
+	n.Namesys = namesys.NewNameSystem(n.Routing)
```

Comment:
It's useful from a systems persective to have less edge-cases. it would be nice to have a routing system always. Can we defer key-loading completely until it's actually _used_ for the first time?  what else does the routing system do? does it load things? does it have to happen then? can it be async?